### PR TITLE
EFM32: add cpu family EFR32ZG23 and board xG23-PK6068A

### PR DESCRIPTION
**Pull request by jue89**
### Contribution description

This PR brings the Gecko series 2 to RIOT:

Current progress of this PR:

- [x] Clock setup
- [x] `periph_gpio`
- [x] `periph_gpio_ll`
- [x] `periph_uart`
- [x] `periph_timer`
- [x] `periph_spi`
- [x] `periph_i2c`
- [x] `periph_hwrng`
- [x] `periph_wdt`
- [x] `periph_flashpage`
- [x] Documentation for the board `xg23a-pk6068a`

I'll come up with a follow-up PR that implements
- `periph_adc`
- `periph_dac`
- `periph_pwm`

### Testing procedure

Murdock and static tests are checking if everything can be built.

#### `examples/default`

* ✓ `periph_uart`
* ✓ `periph_gpio`
* ✓ `periph_i2c` (which is used by `si7021`)

<details>

```
USEMODULE=si7021 make -C examples/default/ BOARD=xg23-pk6068a flash term
> 2022-10-24 21:12:11,951 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:12:11,956 # Welcome to RIOT!
> saul
2022-10-24 21:12:29,712 # saul
2022-10-24 21:12:29,712 # ID	Class		Name
2022-10-24 21:12:29,716 # #0	ACT_SWITCH	LED 0
2022-10-24 21:12:29,716 # #1	ACT_SWITCH	LED 1
2022-10-24 21:12:29,716 # #2	SENSE_BTN	Button 1
2022-10-24 21:12:29,719 # #3	SENSE_BTN	Button 2
2022-10-24 21:12:29,724 # #4	SENSE_TEMP	si70xx
2022-10-24 21:12:29,724 # #5	SENSE_HUM	si70xx
> saul write 0 1
2022-10-24 21:12:42,270 # saul write 0 1
2022-10-24 21:12:42,270 # Writing to device #0 - LED 0
2022-10-24 21:12:42,270 # Data:	              1 
2022-10-24 21:12:42,273 # data successfully written to device #0
(LED0 turns on)
> saul write 1 1
2022-10-24 21:12:47,070 # saul write 1 1
2022-10-24 21:12:47,070 # Writing to device #1 - LED 1
2022-10-24 21:12:47,071 # Data:	              1 
2022-10-24 21:12:47,078 # data successfully written to device #1
(LED1 turns on)
> saul read 2
2022-10-24 21:14:38,190 # saul read 2
2022-10-24 21:14:38,190 # Reading from #2 (Button 1|SENSE_BTN)
2022-10-24 21:14:38,194 # Data:	              0 
(press BTN0)
> saul read 2
2022-10-24 21:14:42,836 # saul read 2
2022-10-24 21:14:42,836 # Reading from #2 (Button 1|SENSE_BTN)
2022-10-24 21:14:42,839 # Data:	              1 
> saul read 3
2022-10-24 21:14:47,096 # saul read 3
2022-10-24 21:14:47,096 # Reading from #3 (Button 2|SENSE_BTN)
2022-10-24 21:14:47,099 # Data:	              0 
(press BTN1)
> saul read 3
2022-10-24 21:14:48,386 # saul read 3
2022-10-24 21:14:48,388 # Reading from #3 (Button 2|SENSE_BTN)
2022-10-24 21:14:48,389 # Data:	              1 
> saul read 4
2022-10-24 21:15:32,415 # saul read 4
2022-10-24 21:15:32,422 # Reading from #4 (si70xx|SENSE_TEMP)
2022-10-24 21:15:32,426 # Data:	          22.93 °C
> saul read 5
2022-10-24 21:15:34,165 # saul read 5
2022-10-24 21:15:34,182 # Reading from #5 (si70xx|SENSE_HUM)
2022-10-24 21:15:34,185 # Data:	          60.75 %
```

</details>

#### `tests/ztimer_msg`

* ✓ `periph_timer` (which is used by `ztimer`)

<details>

```
make -C tests/ztimer_msg/ BOARD=xg23-pk6068a flash term
2022-10-24 21:19:08,278 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:19:08,283 # This is thread 2
2022-10-24 21:19:08,283 # sending 1st msg
2022-10-24 21:19:08,287 # now=2:868472 -> every 2.0s: Hello World
2022-10-24 21:19:08,288 # sending 2nd msg
2022-10-24 21:19:08,290 # now=2:873533 -> every 5.0s: This is a Test
2022-10-24 21:19:08,296 # This is thread 3
2022-10-24 21:19:09,295 # sec=3 min=0 hour=0
2022-10-24 21:19:10,290 # now=4:872121 -> every 2.0s: Hello World
2022-10-24 21:19:10,295 # sec=4 min=0 hour=0
2022-10-24 21:19:11,300 # sec=5 min=0 hour=0
2022-10-24 21:19:12,296 # now=6:875771 -> every 2.0s: Hello World
2022-10-24 21:19:12,301 # sec=6 min=0 hour=0
2022-10-24 21:19:13,295 # now=7:877446 -> every 5.0s: This is a Test
2022-10-24 21:19:13,305 # sec=7 min=0 hour=0
(...)
```

</details>

#### `tests/periph_wdt`

* ✓ `periph_wdt` 

<details>

```
make -C tests/periph_wdt BOARD=xg23-pk6068a flash term
2022-10-24 21:23:28,367 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:23:28,369 # RIOT wdt test application
> range
2022-10-24 21:23:35,109 # range
2022-10-24 21:23:35,114 # lower_bound: 9 upper_bound: 262145 
> setup 0 3000
2022-10-24 21:23:53,948 # setup 0 3000
2022-10-24 21:23:53,948 # valid configuration
> start
2022-10-24 21:23:58,987 # start
2022-10-24 21:23:58,988 # starting wdt timer
> 2022-10-24 21:24:03,132 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:24:03,137 # RIOT wdt test application
> setup 0 3000
2022-10-24 21:24:14,016 # setup 0 3000
2022-10-24 21:24:14,021 # valid configuration
> start
2022-10-24 21:24:36,320 # start
2022-10-24 21:24:36,325 # starting wdt timer
> kick
2022-10-24 21:24:38,615 # kick
2022-10-24 21:24:38,620 # delaying wdt timer
> 2022-10-24 21:24:42,760 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:24:42,765 # RIOT wdt test application
> setup 3000 6000
2022-10-24 21:25:39,219 # setup 3000 6000
2022-10-24 21:25:39,219 # valid configuration
> start
2022-10-24 21:25:42,276 # start
2022-10-24 21:25:42,277 # starting wdt timer
> kick
2022-10-24 21:25:43,866 # kick
2022-10-24 21:25:43,866 # delaying wdt timer
> 2022-10-24 21:25:43,924 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:25:43,926 # RIOT wdt test application
> setup 3000 6000
2022-10-24 21:26:04,915 # setup 3000 6000
2022-10-24 21:26:04,915 # valid configuration
> start
2022-10-24 21:26:07,820 # start
2022-10-24 21:26:07,820 # starting wdt timer
> kick
2022-10-24 21:26:12,409 # kick
2022-10-24 21:26:12,410 # delaying wdt timer
> 2022-10-24 21:26:20,639 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:26:20,639 # RIOT wdt test application
> 
```

</details>

#### `tests/periph_hwrng`

* ✓ `periph_hwrng` 

<details>

```
make -C tests/periph_hwrng BOARD=xg23-pk6068a flash term
2022-10-24 21:29:00,492 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:29:00,492 # 
2022-10-24 21:29:00,492 # HWRNG peripheral driver test
2022-10-24 21:29:00,492 # 
2022-10-24 21:29:00,497 # This test will print from 1 to 20 random bytes about every second
2022-10-24 21:29:00,497 # 
2022-10-24 21:29:00,505 # generating 1 random byte(s)
2022-10-24 21:29:00,505 # Got: 0x84
2022-10-24 21:29:00,505 # generating 2 random byte(s)
2022-10-24 21:29:00,510 # Got: 0x6e 0x9a
2022-10-24 21:29:00,510 # generating 3 random byte(s)
2022-10-24 21:29:00,510 # Got: 0xf5 0xae 0x26
2022-10-24 21:29:00,515 # generating 4 random byte(s)
2022-10-24 21:29:00,515 # Got: 0x4e 0xa6 0x29 0xcf
2022-10-24 21:29:00,522 # generating 5 random byte(s)
2022-10-24 21:29:00,522 # Got: 0x74 0x01 0x3a 0xc6 0x4e
2022-10-24 21:29:00,523 # generating 6 random byte(s)
2022-10-24 21:29:00,527 # Got: 0xb6 0xfe 0x18 0x53 0x73 0x6d
2022-10-24 21:29:00,532 # generating 7 random byte(s)
2022-10-24 21:29:00,535 # Got: 0x96 0x6c 0x1f 0x6d 0x1b 0x40 0xf6
2022-10-24 21:29:00,540 # generating 8 random byte(s)
2022-10-24 21:29:00,540 # Got: 0x34 0x01 0xff 0x9d 0x53 0x00 0x74 0xbe
2022-10-24 21:29:00,547 # generating 9 random byte(s)
2022-10-24 21:29:00,547 # Got: 0xee 0xeb 0xfd 0x57 0x39 0x68 0x65 0x63 0x4a
2022-10-24 21:29:00,557 # generating 10 random byte(s)
2022-10-24 21:29:00,557 # Got: 0xbf 0x78 0x04 0xe9 0xd4 0xad 0x46 0x2f 0xfd 0xaa
2022-10-24 21:29:00,562 # generating 11 random byte(s)
2022-10-24 21:29:00,565 # Got: 0xee 0xc2 0x31 0x97 0x65 0xc3 0x03 0xc5 0x33 0x64 0xa7
2022-10-24 21:29:00,572 # generating 12 random byte(s)
2022-10-24 21:29:00,575 # Got: 0xed 0x53 0x08 0x16 0x75 0xf3 0x33 0xdf 0x43 0xeb 0x07 0x8e
2022-10-24 21:29:00,585 # generating 13 random byte(s)
2022-10-24 21:29:00,586 # Got: 0x95 0xcd 0xb1 0x83 0xae 0xf9 0xfa 0x57 0xc8 0xe3 0x49 0xfb 0x32
2022-10-24 21:29:00,586 # generating 14 random byte(s)
2022-10-24 21:29:00,592 # Got: 0xc9 0x94 0x84 0x36 0x6e 0x5c 0xb8 0x73 0xaa 0x80 0x50 0x0d 0xf6 0xd8
2022-10-24 21:29:00,595 # generating 15 random byte(s)
2022-10-24 21:29:00,602 # Got: 0x27 0x5d 0x61 0xab 0xc1 0x3b 0x7e 0xbd 0xd1 0x89 0x89 0x0d 0x17 0xb2 0x00
2022-10-24 21:29:00,612 # generating 16 random byte(s)
2022-10-24 21:29:00,615 # Got: 0xcf 0x12 0x86 0x83 0x59 0xa3 0xf8 0x42 0x90 0xef 0x56 0x62 0x7d 0xd5 0xdb 0xd0
2022-10-24 21:29:00,622 # generating 17 random byte(s)
2022-10-24 21:29:00,625 # Got: 0xfa 0x1c 0x89 0x5c 0x9b 0xce 0x7e 0x1e 0xfb 0x86 0x30 0x0b 0x95 0x6b 0xa6 0x63 0x7d
2022-10-24 21:29:00,632 # generating 18 random byte(s)
2022-10-24 21:29:00,637 # Got: 0x8f 0x68 0x4d 0x82 0x46 0x7a 0x90 0xd5 0xc8 0x57 0x08 0xe5 0xac 0xfc 0xf1 0x70 0x6b 0xc9
2022-10-24 21:29:00,642 # generating 19 random byte(s)
2022-10-24 21:29:00,650 # Got: 0x77 0x9c 0x06 0xa5 0xc7 0xf7 0xf3 0x79 0xe7 0x51 0x2e 0x1a 0xd1 0xc8 0x9e 0x1d 0xd6 0xab 0xda
2022-10-24 21:29:00,657 # generating 20 random byte(s)
2022-10-24 21:29:00,665 # Got: 0x96 0xa5 0x19 0x54 0x89 0xd5 0xa6 0x10 0xb2 0xf6 0xbc 0x60 0xfc 0xeb 0x95 0x1f 0x3b 0x73 0xa6 0xbc
(...)
```

</details>

#### `tests/periph_gpio_ll` + bridging PC1-PC2 and PC3-PC0

* ✓ `periph_gpio_ll` 

<details>

```
make -C tests/periph_gpio_ll PORT_IN=2 PIN_IN_0=1 PIN_IN_1=3 PORT_OUT=2 PIN_OUT_0=2 PIN_OUT_1=0 BOARD=xg23-pk6068a flash term
2022-10-24 21:35:36,101 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:35:36,105 # Test / Hardware Details:
2022-10-24 21:35:36,105 # ========================
2022-10-24 21:35:36,110 # Cabling:
2022-10-24 21:35:36,110 #   P2.1 (PC1) -- P2.2 (PC2)
2022-10-24 21:35:36,111 #   P2.3 (PC3) -- P2.0 (PC0)
2022-10-24 21:35:36,115 # Number of pull resistor values supported: 1
2022-10-24 21:35:36,120 # Number of drive strengths supported: 1
2022-10-24 21:35:36,125 # Number of slew rates supported: 1
2022-10-24 21:35:36,125 # Valid GPIO ports:
2022-10-24 21:35:36,125 # - PORT 0 (PORT A)
2022-10-24 21:35:36,130 # - PORT 1 (PORT B)
2022-10-24 21:35:36,130 # - PORT 2 (PORT C)
2022-10-24 21:35:36,131 # - PORT 3 (PORT D)
2022-10-24 21:35:36,131 # 
2022-10-24 21:35:36,135 # Testing gpio_port_pack_addr()
2022-10-24 21:35:36,135 # =============================
2022-10-24 21:35:36,135 # 
2022-10-24 21:35:36,136 # All OK
2022-10-24 21:35:36,136 # 
2022-10-24 21:35:36,140 # Testing gpip_ng_init()
2022-10-24 21:35:36,140 # ======================
2022-10-24 21:35:36,140 # 
2022-10-24 21:35:36,146 # Testing is_gpio_port_num_valid() is true for PORT_OUT and PORT_IN:
2022-10-24 21:35:36,146 # 
2022-10-24 21:35:36,150 # Testing input configurations for PIN_IN_0:
2022-10-24 21:35:36,155 # Support for input with pull up: yes
2022-10-24 21:35:36,165 # state: in, pull: up, schmitt trigger: on, value: on
2022-10-24 21:35:36,165 # Support for input with pull down: yes
2022-10-24 21:35:36,166 # state: in, pull: down, schmitt trigger: on, value: off
2022-10-24 21:35:36,170 # Support for input with pull to bus level: no
2022-10-24 21:35:36,175 # Support for floating input (no pull resistors): yes
2022-10-24 21:35:36,180 # state: in, pull: none, schmitt trigger: on, value: off
2022-10-24 21:35:36,180 # 
2022-10-24 21:35:36,185 # Testing output configurations for PIN_OUT_0:
2022-10-24 21:35:36,190 # Support for output (push-pull) with initial value of LOW: yes
2022-10-24 21:35:36,191 # state: out-pp, value: off
2022-10-24 21:35:36,195 # Output is indeed LOW: yes
2022-10-24 21:35:36,200 # state: out-pp, value: on
2022-10-24 21:35:36,200 # Output can be pushed HIGH: yes
2022-10-24 21:35:36,205 # Support for output (push-pull) with initial value of HIGH: yes
2022-10-24 21:35:36,208 # state: out-pp, value: on
2022-10-24 21:35:36,215 # Output is indeed HIGH: yes
2022-10-24 21:35:36,220 # Support for output (open drain with pull up) with initial value of LOW: yes
2022-10-24 21:35:36,230 # state: out-od, pull: up, schmitt trigger: on, value: off
2022-10-24 21:35:36,230 # Output is indeed LOW: yes
2022-10-24 21:35:36,235 # Support for output (open drain with pull up) with initial value of HIGH: yes
2022-10-24 21:35:36,245 # state: out-od, pull: up, schmitt trigger: on, value: on
2022-10-24 21:35:36,245 # Output is indeed HIGH: yes
2022-10-24 21:35:36,250 # Support for output (open drain) with initial value of LOW: yes
2022-10-24 21:35:36,253 # state: out-od, pull: none, schmitt trigger: on, value: off
2022-10-24 21:35:36,260 # Output is indeed LOW: yes
2022-10-24 21:35:36,265 # Support for output (open drain) with initial value of HIGH: yes
2022-10-24 21:35:36,270 # state: out-od, pull: none, schmitt trigger: on, value: on
2022-10-24 21:35:36,275 # state: in, pull: down, schmitt trigger: on, value: off
2022-10-24 21:35:36,275 # Output can indeed be pulled LOW: yes
2022-10-24 21:35:36,285 # state: in, pull: up, schmitt trigger: on, value: on
2022-10-24 21:35:36,285 # Output can indeed be pulled HIGH: yes
2022-10-24 21:35:36,290 # Support for output (open source) with initial value of LOW: yes
2022-10-24 21:35:36,295 # state: out-os, pull: none, schmitt trigger: on, value: off
2022-10-24 21:35:36,305 # state: in, pull: down, schmitt trigger: on, value: off
2022-10-24 21:35:36,306 # Output can indeed be pulled LOW: yes
2022-10-24 21:35:36,315 # state: in, pull: up, schmitt trigger: on, value: on
2022-10-24 21:35:36,315 # Output can indeed be pulled HIGH: yes
2022-10-24 21:35:36,323 # Support for output (open source) with initial value of HIGH: yes
2022-10-24 21:35:36,330 # state: out-os, pull: none, schmitt trigger: on, value: on
2022-10-24 21:35:36,330 # Output is indeed HIGH: yes
2022-10-24 21:35:36,340 # Support for output (open source with pull up) with initial value of HIGH: yes
2022-10-24 21:35:36,340 # Output is indeed HIGH: yes
2022-10-24 21:35:36,350 # Support for output (open source with pull up) with initial value of LOW: yes
2022-10-24 21:35:36,351 # Output is indeed LOW: yes
2022-10-24 21:35:36,355 # Support for disconnecting GPIO: yes
2022-10-24 21:35:36,360 # Output can indeed be pulled LOW: yes
2022-10-24 21:35:36,360 # Output can indeed be pulled HIGH: yes
2022-10-24 21:35:36,361 # 
2022-10-24 21:35:36,365 # Testing Reading/Writing GPIO Ports
2022-10-24 21:35:36,368 # ==================================
2022-10-24 21:35:36,368 # 
2022-10-24 21:35:36,375 # testing initial value of 0 after init
2022-10-24 21:35:36,375 # ...OK
2022-10-24 21:35:36,380 # testing setting both outputs_optional simultaneously
2022-10-24 21:35:36,380 # ...OK
2022-10-24 21:35:36,383 # testing clearing both outputs_optional simultaneously
2022-10-24 21:35:36,388 # ...OK
2022-10-24 21:35:36,388 # testing toggling first output (0 --> 1)
2022-10-24 21:35:36,393 # ...OK
2022-10-24 21:35:36,393 # testing toggling first output (1 --> 0)
2022-10-24 21:35:36,398 # ...OK
2022-10-24 21:35:36,399 # testing toggling second output (0 --> 1)
2022-10-24 21:35:36,403 # ...OK
2022-10-24 21:35:36,408 # testing toggling second output (1 --> 0)
2022-10-24 21:35:36,408 # ...OK
2022-10-24 21:35:36,415 # testing setting first output and clearing second with write
2022-10-24 21:35:36,415 # ...OK
2022-10-24 21:35:36,420 # testing setting second output and clearing first with write
2022-10-24 21:35:36,420 # ...OK
2022-10-24 21:35:36,425 # All input/output operations worked as expected
2022-10-24 21:35:36,425 # 
2022-10-24 21:35:36,425 # 
2022-10-24 21:35:36,425 # TEST SUCCEEDED
```

</details>

#### `tests/periph_flashpage`

* ✓ `periph_flashpage` 

<details>

```
make -C tests/periph_flashpage BOARD=xg23-pk6068a flash term
2022-10-24 21:39:32,902 # main(): This is RIOT! (Version: 2023.01-devel-111-ga824d-feature/efm32-series2)
2022-10-24 21:39:32,903 # ROM flash read write test
2022-10-24 21:39:32,903 # 
2022-10-24 21:39:32,907 # Please refer to the README.md for further information
2022-10-24 21:39:32,908 # 
2022-10-24 21:39:32,908 # Flash start addr:		0x08000000
2022-10-24 21:39:32,912 # Page size:			8192
2022-10-24 21:39:32,913 # Number of pages:		64
2022-10-24 21:39:32,915 # Number of first free page: 	6 
2022-10-24 21:39:32,920 # Number of last free page: 	63 
> dump 63
2022-10-24 21:40:06,460 # dump 63
2022-10-24 21:40:06,461 # Flash page 63 at address 0x807e000
2022-10-24 21:40:06,471 # 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 
2022-10-24 21:40:06,475 #   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ?? 
2022-10-24 21:40:06,481 # 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 
2022-10-24 21:40:06,490 #   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ?? 
(...)
> test 63
2022-10-24 21:41:24,429 # test 63
2022-10-24 21:41:24,549 # wrote local page buffer to flash page 63 at addr 0x807e000
> dump 63
2022-10-24 21:41:42,473 # dump 63
2022-10-24 21:41:42,480 # Flash page 63 at address 0x807e000
2022-10-24 21:41:42,481 # 0x61 0x62 0x63 0x64 0x65 0x66 0x67 0x68 0x69 0x6a 0x6b 0x6c 0x6d 0x6e 0x6f 0x70 
2022-10-24 21:41:42,495 #   a    b    c    d    e    f    g    h    i    j    k    l    m    n    o    p  
2022-10-24 21:41:42,496 # 0x71 0x72 0x73 0x74 0x75 0x76 0x77 0x78 0x79 0x7a 0x61 0x62 0x63 0x64 0x65 0x66 
2022-10-24 21:41:42,508 #   q    r    s    t    u    v    w    x    y    z    a    b    c    d    e    f  
(...)
> erase 63
2022-10-24 21:42:09,552 # erase 63
2022-10-24 21:42:09,567 # successfully erased page 63 (addr 0x807e000)
> dump 63
2022-10-24 21:42:21,806 # dump 63
2022-10-24 21:42:21,813 # Flash page 63 at address 0x807e000
2022-10-24 21:42:21,814 # 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 
2022-10-24 21:42:21,821 #   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ?? 
2022-10-24 21:42:21,829 # 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 
2022-10-24 21:42:21,843 #   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ??   ?? 
(...)
```

</details>

### Issues/PRs references

Follow up of #18760

Based on:
- #18739
- #18764

### CODE REVIEW

File: dist/tools/doccheck/exclude_patterns [hunk 1]
```diff
@@ -13110,3 +13110,41 @@ boards/sltb009a/include/periph_conf\.h:[0-9]+: warning: Member TIMER_[A-Z0-9_]+
 boards/sltb009a/include/periph_conf\.h:[0-9]+: warning: Member timer_config\[\] \(variable) of file periph_conf\.h is not documented.
 boards/sltb009a/include/periph_conf\.h:[0-9]+: warning: Member UART_[A-Z0-9_]+ \(macro definition) of file periph_conf\.h is not documented.
 boards/sltb009a/include/periph_conf\.h:[0-9]+: warning: Member uart_config\[\] \(variable) of file periph_conf\.h is not documented.
+boards/xg23b\-pk6068a/include/board\.h:[0-9]+: warning: Member CONFIG_ZTIMER_LPTIMER_DEV \(macro definition\) of file board\.h is not documented\.
```
Comment (gschorcht) [inline dist/tools/doccheck/exclude_patterns:13113]: Number of entries could be reduced a lot by using regular expressions for identifiers, for example
```suggestion
boards/xg23b\-pk6068a/include/board\.h:[0-9]+: warning: Member CONFIG_ZTIMER_[A-Z_]+ \(macro definition\) of file board\.h is not documented\.
```

Comment (jue89) [reply dist/tools/doccheck/exclude_patterns:13113]: 5dfa95795953a61cc33281b095196a9ef776fc07

File: dist/tools/doccheck/exclude_patterns [hunk 2]
```diff
@@ -13110,3 +13110,27 @@ boards/sltb009a/include/periph_conf\.h:[0-9]+: warning: Member TIMER_[A-Z0-9_]+
 boards/sltb009a/include/periph_conf\.h:[0-9]+: warning: Member timer_config\[\] \(variable) of file periph_conf\.h is not documented.
 boards/sltb009a/include/periph_conf\.h:[0-9]+: warning: Member UART_[A-Z0-9_]+ \(macro definition) of file periph_conf\.h is not documented.
 boards/sltb009a/include/periph_conf\.h:[0-9]+: warning: Member uart_config\[\] \(variable) of file periph_conf\.h is not documented.
+boards/xg23b\-pk6068a/include/board\.h:[0-9]+: warning: Member CONFIG_ZTIMER_[A-Z_]+ \(macro definition\) of file board\.h is not documented\.
+boards/xg23b\-pk6068a/include/board\.h:[0-9]+: warning: Member VCOM_[A-Z_]+ \(macro definition\) of file board\.h is not documented\.
+boards/xg23b\-pk6068a/include/board\.h:[0-9]+: warning: Member PB0_PIN \(macro definition\) of file board\.h is not documented\.
+boards/xg23b\-pk6068a/include/board\.h:[0-9]+: warning: Member PB1_PIN \(macro definition\) of file board\.h is not documented\.
```
Comment (gschorcht) [inline dist/tools/doccheck/exclude_patterns:13116]:
```suggestion
boards/xg23b\-pk6068a/include/board\.h:[0-9]+: warning: Member PB[0-1]_PIN \(macro definition\) of file board\.h is not documented\.
```
:wink:

Comment (jue89) [reply dist/tools/doccheck/exclude_patterns]: 2a560013ca9de4f82ab3dd36052366b70fc1331d

**Comment by gschorcht** [comment]
Code looks good, only the documentation is missing from my point of view. Feell free to squash when you provide the changes in `doc.txt`.

File: boards/xg23-pk6068a/doc.txt [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+| PC2  | 6   | 5   | PA0  |
+| PC1  | 4   | 3   | PA10 |
+| VMCU | 2   | 1   | GND  |
+
+
+### Peripheral mapping
+| Peripheral | Number  | Hardware        | Pins                           | Comments                         | Min. Power Mode |
+|------------|---------|-----------------|--------------------------------|----------------------------------|-----------------|
+| I2C        | 0       | I2C0            | SDA: PC7,  CLK: PC5            |                                  | EM1             |
+| SPI        | 0       | USART0          | MOSI: PC1, MISO: PC2, CLK: PC3 |                                  | EM1             |
```
[the remark this answers is not in the document]
Comment (jue89) [reply boards/xg23-pk6068a/doc.txt]: Thank you! Added and squashed right away.

**Review by gschorcht — APPROVED** [review]

**Comment by jue89** [comment]
Thank you a lot for your review!

**Pull request merged**